### ISSUE TITLE
Add functionality to support the definition of subset selections to the R-wrapper

### ISSUE THREAD

**Issue by ArnoKnobbe** [SubDisc/SubDisc#27]
The R-wrapper is almost complete. What is still missing is the option to specify a subset (selection of the data). This allows you to run SD not on the entire dataset but on a selection, specified by a single condition on one of the available attributes. Two tests in EndToEndTest.java demonstrate how this works in practice: testAdult8 and testAdult9. Most of the code there is standard. I indicated the specific part of the code that needs support in the R-wrapper.

**Comment by JGobeil** [comment]
I'm looking at the examples. Is it like doing the subgroup discovery using only a subset of the data?

**Comment by ArnoKnobbe** [comment]
Yes, that's all there is to it. So the extra functionality for the wrapper is to allow the specification of the subset.

**Comment by JGobeil** [comment]
This is something that R is quite good at. So went passing the data directly to R, the simple way is to do those kinds of transformations before passing the data.

When passing a file to load to the wrapper I see three approaches.
0. The user cannot. If it wants to pass a subset, it needs to load the data in R and calculate the subset in R.
1. Adding an optional parameter to the wrapper that tells the "selection" (an array of boolean). 
2. Adding multiple optional parameters for the target columns, the target value and the kind of comparison. 

Option 0 is already there if needed.

Option 1 is simple to implement and, like option 0, offers a lot of flexibility on how to create the subset as the full power of R for that kind of task is there. However, this option implies that the data is first loaded in R to create the "selection". At that point is simpler to just pass the subset directly. 

Option 2 need more parameters, work and testing, and is less flexible.

In my opinion, I would say that option 0 is reasonable (not only because it is less work). R is quite powerful to create subsets (see for example the function `filter` https://dplyr.tidyverse.org/reference/filter.html).  You can easily create complex subsets combining multiple parameters. It is not unreasonable to ask if a user wants to use a subset of a dataset to process it in R before passing it to the wrapper.

What do you think?

**Comment by ArnoKnobbe** [comment]
Well, perhaps. But you have to keep in mind that this subset 'creation' will have to be done thousands of times in a row on the same dataset. For example, if you have a database of all people in the Netherlands, then they want an SD run on each individual municipality with its inhabitants. So the subset will be created on the attribute 'municipality' with each of the possible values as one selector. After each specification, you run SD, store the results and move to the next.
Of course, this can be done in R also, but you would extract a subset thousands of times in a row, and keep the original dataset in memory at the same time. For a single selection, your suggestion makes sense, but for the repeated version, it's a bit cumbersome. But it can be done, yes. But the functionality is available (and essential) in Cortana (the GUI), so the users would want to use it through the wrapper also.
Just to make sure you understand, there is no new functionality required. This functionality already exists and is tested. It just needs to be exposed through the wrapper.
The functionality is actually quite efficient under the hood, since it doesn't copy any data, it just keeps a bitstring of members, which is very similar to how a subgroup works.

**Comment by ArnoKnobbe** [comment]
Hi Jeremy, I noticed you are no longer using Slack through LIACS. Let's see if this reaches you:)
Have you thought about my comment above. I guess we can proceed with the subset selection through R as you suggested. But if you can find the time, I would appreciate you including the functionality in the wrapper since that would potentially be the most efficient implementation. But I also understand if you don't have time or see the urgency.
Thanks
